File: tests/nodejs/runner.js
(function(context) {
  'use strict';

  var path = require('path')

  // Kind of namespace for classes
  var IOBench = (function() {
    return {
      read: require('./tests/read.js'),
      write: require('./tests/write.js'),
      parse: require('./tests/parse.js')
    }
  })();


  var init = (function(args) {
    var file_path = args.TEST_FILE,
        test_type = args.TEST_TYPE,
        test_mode = (args.TEST_MODE || 'full'),
        test_name = (args.TEST_NAME || path.basename(file_path)),
        verbose = (args.VERBOSE == 'true');

    var bench;

    if(test_type == "" || test_type == null) {
      throw('You must suply TEST_TYPE with one the values: read, write, read+write')
    }

    var options = {}

    if(test_type == 'read') {
      bench = new IOBench.read
    }
    else if(test_type == 'parse') {
      bench = new IOBench.parse
    } else if(test_type == 'write') {
      var block_size = parseInt(args.KB_BLOCK_SIZE || 1)
      options.block_size = block_size

      bench = new IOBench.write
    }

    if(bench) {
      bench.setup(file_path, test_mode, test_name, test_type, verbose, options)
      bench.run()
    }
  });

  init(process.env);

})(this)

File: tests/ruby/runner.rb
require 'pry'

require_relative 'tests/read'
require_relative 'tests/write'
require_relative 'tests/parse'

module IOBench
  module Runner

    def self.start(env)
      file_path = env['TEST_FILE']
      test_type = env['TEST_TYPE']
      test_mode = (env['TEST_MODE'] || 'full')
      test_name = (ENV['TEST_NAME'] || File.basename(file_path))
      verbose   = (env['VERBOSE'] == 'true')

      args = [file_path, test_mode, test_name, verbose]

      raise('You must suply TEST_TYPE with one the values: read, write, read+write') if test_type.nil?

      if test_type == 'write'
        # avoid override if not test file is given for write tests
        # args[0] = "#{args[0]}.write" if File.exists?(args[0])
        args.unshift (env['KB_BLOCK_SIZE'] || 1).to_i
      end

      klass = test_type.downcase.capitalize
      bench = IOBench.const_get(klass).new(*args)
      bench.run
    end
  end
end


IOBench::Runner.start(ENV)